apps/server/src/modules/user-login-migration/uc/user-login-migration.uc.spec.ts

```diff
@@ -626,188 +627,510 @@ describe(UserLoginMigrationUc.name, () => {
 	});
 
 	describe('forceMigration', () => {
-		describe('when the user and their school can successfully migrate', () => {
-			const setup = () => {
-				const user = userFactory.buildWithId();
-				const userDo = userDoFactory.build({
-					id: user.id,
-					roles: [
-						new RoleReference({
-							id: new ObjectId().toHexString(),
-							name: RoleName.ADMINISTRATOR,
-						}),
-					],
-				});
-				const externalUserId = 'externalUserId';
-				const externalSchoolId = 'externalSchoolId';
-				const school = legacySchoolDoFactory.build({
-					id: user.school.id,
-				});
-				const userLoginMigration = userLoginMigrationDOFactory.buildWithId({
-					schoolId: user.school.id,
+		describe('when the school is not migrated with no active user login migration', () => {
```

Comment:
this overarching describe is actually not needed, because there is only 1 describe beneath it. you could merge them into one describe save yourself one layer :) The setupNonMigratedSchool would be used in the normal setup because it is only called once.